### PR TITLE
Add GH Workflow to run php-cs-fixer on PRs and PUSH to master

### PR DESCRIPTION
**Pull request by dlondero**
Refs #19 

@alekitto this should cover the case of running `php-cs-fixer` on pull requests and direct pushed to master. Runs it using the `.php_cs.dist` and auto commits changes (if any) to the branch where it is running.

### CODE REVIEW

File: .github/workflows/php-cs-fixer.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  php-cs-fixer:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install
+        run: composer install
```
Comment (alekitto) [inline .github/workflows/php-cs-fixer.yml]: Maybe `- uses: php-actions/composer@v2` could be more effective here?

**Review by alekitto — COMMENTED** [review]
Hi @dlondero!
Thank you for your work.

Just a few considerations:

- probably `pull_request_target` should be added to the `on` section to make this work for PR from forks.
- I'd prefer to run the check only on `master`, without autocommit (@massimilianobraglia what do you think about it?)

Until this is not merged, this workflow will not run. Have you tried to enable actions on your fork to test it? Or do you know a way to run it before this is merged?

File: .github/workflows/php-cs-fixer.yml (hunk 1 above)
Comment (dlondero) [reply .github/workflows/php-cs-fixer.yml]: Sgtm

**Comment by dlondero** [comment]
> * I'd prefer to run the check only on `master`, without autocommit (@massimilianobraglia what do you think about it?)

You mean to run it only on PRs targeting `master` branch and just warn the contributor about the CS fixes they still need to push?

**Comment by alekitto** [comment]
> You mean to run it only on PRs targeting `master` branch and just warn the contributor about the CS fixes they still need to push?

No, what I mean is to run the fixer and use the `git-auto-commit-action` to automatically push the fixes on PRs (any target), but to run the check with no fix on `master` branch (the workflow must fail and report the failing status).

**Comment by dlondero** [comment]
`git-auto-commit-action` step could be excluded checking the `base_ref` probably.
Not sure to understand the `no fix` to run on master: what command would you run? Something like `vendor/bin/php-cs-fixer fix --dry-run` and see how to mark the workflow as failed somehow?

Probably worth having 2 separate workflows then?

**Comment by alekitto** [comment]
> Not sure to understand the `no fix` to run on master: what command would you run? Something like `vendor/bin/php-cs-fixer fix --dry-run` and see how to mark the workflow as failed somehow?

`--dry-run` should exit with non-zero code if one or more files needs to be fixed, stopping the build with failed status.

> Probably worth having 2 separate workflows then?

I thought the same, maintenance should be easier with two separate workflows.

**Comment by dlondero** [comment]
@alekitto I did some more tests in order to have the workflow to run when working with forked repos. I found out that the `git-auto-commit-action` [would not work](https://github.com/stefanzweifel/git-auto-commit-action#using-the-action-in-forks-from-public-repositories) for forked repos if not enabling GH Actions also on the forked repo and ultimately the workflow would run there and not on the origin repo. That's why they suggest to have it running when pushing on `master` (or default branch).

How to proceed then? Just `--dry-run` on both PRs and push on `master` and report the failing status?

**Comment by dlondero** [comment]
We could also pass more flags `--diff --diff-format=udiff --verbose` so if someone checks why the action failed they can see what's going on.

**Comment by massimilianobraglia** [comment]
Hey there!
Thank you for your time, @dlondero!

> * I'd prefer to run the check only on `master`, without autocommit (@massimilianobraglia what do you think about it?)

Yes, cs-fixer's dry-run on master and autocommit on PRs would be ideal IMO. 

> @alekitto I did some more tests in order to have the workflow to run when working with forked repos. I found out that the `git-auto-commit-action` [would not work](https://github.com/stefanzweifel/git-auto-commit-action#using-the-action-in-forks-from-public-repositories) for forked repos if not enabling GH Actions also on the forked repo and ultimately the workflow would run there and not on the origin repo. That's why they suggest to have it running when pushing on `master` (or default branch).
> 
> How to proceed then? Just `--dry-run` on both PRs and push on `master` and report the failing status?

I did not understand well, sorry about this. Is the action triggered on the PR? If not this is a problem, otherwise we got that.
If someone wants to do some tests on his/her fork is ok, but we need to remind them that actions must be enabled (maybe something on README.md could be nice?)

> We could also pass more flags `--diff --diff-format=udiff --verbose` so if someone checks why the action failed they can see what's going on.

Good point. We can check later if it is too much verbose or not 😄

**Comment by dlondero** [comment]
> I did not understand well, sorry about this. Is the action triggered on the PR? If not this is a problem, otherwise we got that.
> If someone wants to do some tests on his/her fork is ok, but we need to remind them that actions must be enabled (maybe something on README.md could be nice?)

@massimilianobraglia workflow works on PRs as well running the fixer on the code from the forked repo. What can't be accomplished there is the auto-commit with the fixes (if any) because of the reasons reported previously. 

So in the end both workflows (master and PRs) would do a `--dry-run` and alert (failing) a contributor that there's something that needs to be cs-fixed.

**Comment by massimilianobraglia** [comment]
LGTM. @alekitto what do you think about it?

**Review by alekitto — APPROVED** [review]
Sorry for the delay. LGTM too. 👍 

I think this is the only way for now. I'll search a way to auto-push code style fixes on PRs in the future, but this check is enough for now.

**Comment by alekitto** [comment]
Thanks again for your work @dlondero

**Pull request merged**